src/reanimated2/hook/utils.ts

```diff
@@ -42,8 +42,10 @@ export function areDependenciesEqual(
   prevDeps: DependencyList
 ) {
   function is(x: number, y: number) {
-    // eslint-disable-next-line no-self-compare
-    return (x === y && (x !== 0 || 1 / x === 1 / y)) || (x !== x && y !== y);
+    return (
+      (x === y && (x !== 0 || 1 / x === 1 / y)) ||
```

Comment:
What is the purpose of this code? Handling `Infinity`? Could we compare directly x with infinity to make it easier to read instead?

Reply:
I'm actually not sure either what do we achieve with
`(x !== 0 || 1 / x === 1 / y)`.

Reply:
Okay, I know the answer. In JavaScript the number type has values for `-0` and `0`. However, `-0 === 0` is `true`. So if we want to check if we have this case we need to compare their reciprocals since `1/0` gives `Number.POSITIVE_INFINITY` and `1/(-0)` gives `NUMBER.NEGATIVE_INFINITY` and `Number.POSITIVE_INFINITY === Number.NEGATIVE_INFINITY` is `false`.

Reply:
Why did you undo all this changes? Could you add a comment explaining this code (since it was unintuitive for me)?

Reply:
I didn't undo them, I forgot to merge this branch into current main and it had stale changes.